gnomad_qc/v4/assessment/logistic_regression.py

```diff
@@ -0,0 +1,179 @@
+
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad_qc.v4.resources.basics import get_gnomad_v4_genomes_vds, get_gnomad_v4_vds
+from gnomad_qc.v4.resources.release import release_sites
+from gnomad_qc.v4.resources.sample_qc import ancestry_pca_scores
+
+# Steps to perform logistic regression:
+# 1. Load gnomAD v4 exomes and genomes VDS, and split_multiallelics
+# 2. Union exomes and genomes matrix tables
+# 3. Filter and densify the v3 genomes matrix table
+# 4. Filter to the test lists
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def get_test_intervals(ht: hl.Table) -> hl.Table:
+    """
+    Get the test lists for the exomes and genomes MatrixTables.
+
+    :param joint_ht: Joint HT of v4 exomes and genomes.
+    :return: Test Table
+    """
+    # Filter to chr22
+    ht = hl.filter_intervals(ht, [hl.parse_locus_interval("chr22")])
+
+    # Filter to variants with AF >= 0.1 and p-value < 10e-4
+    ht1 = ht.filter(
+        (ht.joint.freq[0].AF >= 0.1)
+        & (ht.freq_comparison_stats.stat_union.p_value < 10e-4)
+        & (
+            ht.freq_comparison_stats.stat_union.stat_test_name
+            == "cochran_mantel_haenszel_test"
+        )
+    )
+    # Filter to variants with AF < 0.1 and p-value < 10e-4
+    ht2 = ht.filter(
+        (ht.joint.freq[0].AF < 0.1)
+        & (ht.joint.freq[0].AC > 0)
+        & (ht.freq_comparison_stats.stat_union.p_value < 10e-4)
+        & (
+            ht.freq_comparison_stats.stat_union.stat_test_name
+            == "cochran_mantel_haenszel_test"
+        )
+    )
+    ht2 = ht2.naive_coalesce(50)
+    ht2 = ht2._filter_partitions(range(5))
+    # Filter to ~8000 variants with p-value >= 10e-4
+    ht3 = ht.filter(
+        (ht.freq_comparison_stats.stat_union.p_value >= 10e-4)
+        & (
+            ht.freq_comparison_stats.stat_union.stat_test_name
+            == "cochran_mantel_haenszel_test"
+        )
+    )
+    ht3 = ht3._filter_partitions(range(5))
+    ht = ht1.union(ht2).union(ht3)
+
+    # make this an interval table
+    ht = ht.annotate(
+        interval=hl.locus_interval(
+            ht.locus.contig, ht.locus.position, ht.locus.position + 1
+        )
+    )
+    ht = ht.key_by(ht.interval)
+    ht = ht.select()
+
+    return ht
+
+
+def densify_union_exomes_genomes(
+    exomes_vds: hl.vds.VariantDataset,
+    genomes_vds: hl.vds.VariantDataset,
+) -> hl.MatrixTable:
+    """
+    Union exomes and genomes MatrixTables.
+
+    :param exomes_vds: VDS of exomes
+    :param genomes_vds: VDS of genomes
+    :return: Unioned MatrixTable
+    """
+    logger.info("Densifying exomes...")
+    exomes_mt = hl.vds.to_dense_mt(exomes_vds)
+    exomes_mt = exomes_mt.annotate_cols(is_genome=False)
+    exomes_mt = exomes_mt.select_entries("GT").select_rows().select_cols("is_genome")
```

Comment:
If this is the only entry you need then you should filter to it before the densify. I think for this you probably also want to filter to only adj genotypes though, so you probably need more

Reply:
I agree, adj seems to make more sense. I will get that.

Reply:
I found the entries are not the same as what we used in generate freq for exomes or genomes, could you check the new steps for me?